src/pages/Account/Tabs/ModulesTab/ViewCode.tsx

```diff
@@ -67,44 +70,68 @@ function ViewCode({address}: {address: string}): JSX.Element {
     "0x1::code::PackageRegistry",
   );
 
-  const [selectedModuleIndex, setSelectedModuleIndex] =
-    React.useState<number>(0);
-
   const packages: PackageMetadata[] =
     registry === undefined ? [] : (registry.data as any).packages;
   const modules = orderBy(
     packages.flatMap((p) => p.modules),
     "name",
   );
 
+  const navigate = useNavigate();
+
+  const selectedModuleName = useParams().selectedModuleName ?? "";
+  if (!selectedModuleName && modules.length > 0) {
+    navigate(`/account/${address}/modules/${modules[0].name}`, {replace: true});
+  }
+
   if (modules.length === 0) {
     return <EmptyTabContent />;
   }
 
+  const selectedModule = modules.find(
+    (module) => module.name === selectedModuleName,
+  );
+
+  function getLinkToModule(moduleName: string) {
+    return `/account/${address}/modules/${moduleName}`;
+  }
+
+  function navigateToModule(moduleName: string) {
+    navigate(getLinkToModule(moduleName));
+  }
+
   return (
     <Grid container spacing={2}>
       <Grid item md={3} xs={12}>
         <ModuleSidebar
           moduleNames={modules.map((m) => m.name)}
-          selectedModuleIndex={selectedModuleIndex}
-          setSelectedModuleIndex={setSelectedModuleIndex}
+          selectedModuleName={selectedModuleName}
+          getLinkToModule={getLinkToModule}
+          navigateToModule={navigateToModule}
         />
       </Grid>
       <Grid item md={9} xs={12}>
-        <ModuleContent
-          address={address}
-          moduleName={modules[selectedModuleIndex].name}
-          bytecode={modules[selectedModuleIndex].source}
-        />
+        {selectedModule === undefined ? (
+          <EmptyTabContent
+            message={`No module found with name: ${selectedModuleName}`}
+          />
+        ) : (
+          <ModuleContent
+            address={address}
+            moduleName={selectedModuleName}
+            bytecode={selectedModule.source}
+          />
+        )}
       </Grid>
     </Grid>
   );
 }
 
 function ModuleSidebar({
```

Comment:
lets also sync with Lex on this. rn its not very intuitive that users can click on module name to open up another tab. having an icon next to it might be more helpful

Reply:
I think that's standard behaviour of browser links.
Take for example Github - I can right click in a lot of places to open something in a new tab (or window as desired):
image.png
It's a feature, not a bug 😄